posts/2020/03/pytorch-tensoriterator-internals.md

```diff
@@ -0,0 +1,236 @@
+<!--
+.. title: PyTorch TensorIterator Internals
+.. slug: pytorch-tensoriterator-internals
+.. date: 2020-03-12 22:39:56 UTC-05:00
+.. tags: 
+.. category: 
+.. link: 
+.. description: 
+.. type: text
+-->
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-generate-toc again -->
+**Table of Contents**
+
+- [Introduction](#introduction)
+- [History of TensorIterator](#history-of-tensoriterator)
+    - [TH iterators](#th-iterators)
+    - [Limitations of TH iterators](#limitations-of-th-iterators)
+- [Basics of TensorIterator](#basics-of-tensoriterator)
+- [Performing iterations](#performing-iterations)
+    - [Iteration details](#iteration-details)
+        - [Setting tensor iteration dimensions](#setting-tensor-iteration-dimensions)
+- [Conclusion](#conclusion)
+
+<!-- markdown-toc end -->
+
+# Introduction
+
+Pytorch is a massive
+codebase (approx. 12 GB after taking into account the build and generated files), and having
```

Comment:
how about adding "almost a million lines of C++, Python and CUDA code, and approx. 12 GB ..."

got the number from https://www.openhub.net/p/pytorch